types/yup/index.d.ts

```diff
@@ -212,18 +212,29 @@ export type TransformFunction<T> = ((
     originalValue: any
 ) => any);
 
-export interface WhenOptionsBuilder<T> {
+export interface WhenOptionsBuilderFunction<T> {
     (value: any, schema: T): T;
     (v1: any, v2: any, schema: T): T;
     (v1: any, v2: any, v3: any, schema: T): T;
     (v1: any, v2: any, v3: any, v4: any, schema: T): T;
 }
 
-export type WhenOptions<T> =
-    | WhenOptionsBuilder<T>
-    | { is: boolean | ((...values: any[]) => boolean); then: any; otherwise: any }
+export type WhenOptionsBuilderObjectIs =
+    | ((...values: any[]) => boolean)
+    | boolean
+    | number
+    | null
+    | object
+    | string;
+
+export type WhenOptionsBuilderObject =
```

Comment:
We should just call this `WhenOptionsBuilder` to match the documentation for yup.

Reply:
Referring back to the 1st comment. `builder` can be an object or function, so suffixing them makes sense to me to distinguish them.